src/adapters/github.js

```diff
@@ -161,7 +162,57 @@ class GitHub extends PjaxAdapter {
 
     const repo = {username, reponame, branch, pullNumber};
     if (repo.branch) {
-      cb(null, repo);
+      if (isCodePage && type && type !== 'commit') {
+        // The window.location.pathname would be one of below forms:
+        //  - /username/repo/tree (or blob)/branch1
+        //  - /username/repo/tree (or blob)/branch1/folder1
+        //  - /username/repo/tree (or blob)/features/a (Gitflow workflow)
+        //  - /username/repo/tree (or blob)/features/a/folder1
+        // Reserve the branch name which is possibly mixing with the file path
+        const typeIdWithPath = match.slice(3);
+        let repoBranches, repoTags;
+
+        this._get('/branches', {repo, token}, (err, data) => {
+          repoBranches = err ? [] : data.map((item) => (item.name));
+
+          onResponseAvailable(err);
+        });
+
+        this._get('/tags', {repo, token}, (err, data) => {
+          repoTags = err ? [] : data.map((item) => (item.name));
+
+          onResponseAvailable(err);
+        });
+
+        function onResponseAvailable(err) {
+          if (!repoBranches || !repoTags) return;
+
+          if (err) return cb(err);
+
+          const gitSnapshotNames = repoBranches.concat(repoTags);
+
+          let branchNameToTry = '';
+          let foundBranch = false;
+
+          // Verify the branch / tag name by comparing with all branches and tags from the repo
+          typeIdWithPath.forEach((typeIdPart, index) => {
```

Comment:
I don't get this. Isn't the first element of `typeIdWithPath` the branch/tag already? What is this loop about?